Application/Features/Queries/ShowAvailableReservations/ShowAvailableReservationsQueryHandler.cs

```diff
@@ -36,15 +36,18 @@ public class ShowAvailableReservationsQueryHandler : IRequestHandler<ShowAvailab
 
         var availableReservations = new List<AvailableReservationDto>();
 
-        foreach (var reservation in reservations)
+        foreach (var movie in movies)
         {
-            var movie = movies.FirstOrDefault(x => x.Id == reservation.MovieId);
-            if (movie is not null)
+            var reservationList = reservations.FindAll(x => x.MovieId == movie.Id);
+            if (reservationList.Count > 0)
             {
-                availableReservations.Add(AvailableReservationDto.FromMovieReservation(movie, reservation));
+                availableReservations.AddRange(reservationList.Select(reservation => AvailableReservationDto.FromMovieReservation(movie, reservation)));
             }
         }
 
-        return availableReservations;
+        return availableReservations
+            .Skip((request.PageNumber - 1) * request.PageSize)
```

Comment:
Ovo -1 je dosta čudno - deluje kao neki magic number. Ili da se objasni što je tako (tako što se to -1 izdvoji u varijablu sa imenom koje opisuje svrhu), ili da se reši na neki način da se ne koristi uopšte